src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -102,7 +103,16 @@ void ReadHeader()
         /// The number of columns in the row. -1 if there are no further rows.
         /// Note: This will currently be the same value for all rows, but this may change in the future.
         /// </returns>
-        public int StartRow()
+        public int StartRow() => StartRow(false).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Starts reading a single row, must be invoked before reading any columns.
+        /// </summary>
+        /// <returns>
+        /// The number of columns in the row. -1 if there are no further rows.
+        /// Note: This will currently be the same value for all rows, but this may change in the future.
+        /// </returns>
+        public async Task<int> StartRow(bool async)
```

Comment:
This should be a `ValueTask<int>`, since in most cases data is already buffered in memory and the method will complete synchronously, without any I/O. This is the case with all methods which return a value (i.e. Task of something), as opposed to methods which don't return and which can return Task - the latter can simply return `Task.Completed` when returning synchronously.